pkg/controllers/jobset_controller.go

```diff
@@ -701,6 +704,10 @@ func GenSubdomain(js *jobset.JobSet) string {
 	return js.Name
 }
 
+func namespacedJobName(ns string, jobName string) string {
```

Comment:
pass the jobset itself, the object include the name and namespace

Reply:
I originally did this until I started updating the unit tests and realized the [makeJob](https://github.com/kubernetes-sigs/jobset/blob/main/pkg/controllers/jobset_controller_test.go#L1011-L1021) test util function does not accept the JobSet itself as one of the params, but rather the makeJobArgs struct with various parameters to set for the job. So I had to switch to the current function signature to be compatible from calls within the controller and in the unit tests.

Reply:
replaced this with jobHashKey function